src/merkle_reg.rs

```diff
@@ -122,6 +122,11 @@ impl<T> MerkleReg<T> {
         self.dag.get(&hash).or_else(|| self.orphans.get(&hash))
     }
 
+    /// Iterator over all the nodes in the Merkle DAG.
+    pub fn all_nodes(&self) -> impl Iterator<Item = &Node<T>> {
+        self.dag.values().chain(self.orphans.values())
```

Comment:
Don't include the orphans, those are not validated so should not be considered as part of the DAg

Reply:
Ah! Got it, made the changes..